lib/microformats/format_parser.rb

```diff
@@ -347,10 +122,262 @@ def check_for_h_properties
+        end
+
+        if child_a_tags_with_href.count == 1
+          node = child_a_tags_with_href.first
+
+          if format_classes(node).empty?
+            @properties['url'] = [node.attribute('href').value.strip]
+          end
+        end
+
+        if @properties['url'].nil?
+          # else if .h-x>area[href]:only-of-type:not[.h-*], then use that [href] for url
+          child_area_tags_with_href = element.children.select do |child|
+            child.is_a?(Nokogiri::XML::Element) && child.name == 'area' && !child.attribute('href').nil?
+          end
+
+          if child_area_tags_with_href.count == 1
+            node = child_area_tags_with_href.first
+
+            if format_classes(node).empty?
+              @properties['url'] = [node.attribute('href').value.strip]
+            end
+          end
+        end
+
+        child_elements = element.children.reject { |child| child.is_a?(Nokogiri::XML::Text) }
+
+        if @properties['url'].nil? && child_elements.count == 1 && format_classes(child_elements.first).empty?
+          child_element = child_elements.first
+          # else if .h-x>:only-child:not[.h-*]>a[href]:only-of-type:not[.h-*], then use that [href] for url
+          child_a_tags_with_href = child_element.children.select do |child|
+            child.is_a?(Nokogiri::XML::Element) && child.name == 'a' && !child.attribute('href').nil?
+          end
+
+          if child_a_tags_with_href.count == 1
+            node = child_a_tags_with_href.first
+
+            if format_classes(node).empty?
+              @properties['url'] = [node.attribute('href').value.strip]
+            end
+          end
+
+          if @properties['url'].nil?
+            # else if .h-x>:only-child:not[.h-*]>area[href]:only-of-type:not[.h-*], then use that [href] for url
+            child_area_tags_with_href = child_element.children.select do |child|
+              child.is_a?(Nokogiri::XML::Element) && child.name == 'area' && !child.attribute('href').nil?
+            end
+
+            if child_area_tags_with_href.count == 1
+              node = child_area_tags_with_href.first
+
+              if format_classes(node).empty?
+                @properties['url'] = [node.attribute('href').value.strip]
+              end
+            end
+          end
+        end
+      end
+
+      unless @properties['url'].nil?
+        @properties['url'] = [Microformats::AbsoluteUri.new(@properties['url'].first, base: @base).absolutize]
+      end
+    end
+
+    def prefix_from_class(class_name)
+      class_name.downcase.split('-')[0]
+    end
+
+    def property_from_class(class_name)
+      class_name.downcase.split('-')[1..-1].join('-')
+    end
+
+    def imply_dates(element)
+      ### imply date for dt-end if dt-start is defined with a date ###
+      if !@properties['end'].nil? && !@properties['start'].nil?
+        start_date = nil
+
+        @properties['start'].each do |start_val|
+          if start_val =~ /^(\d{4}-[01]\d-[0-3]\d)/
```

Comment:
I think `=~` here and on line 350 can be `.match?` instead.

Reply:
it cannot be because the Regexp.last_match would fail then.
could convert it to
`if matchdata = start_val.match(...)`
but i really hate using = in an if test

Reply:
Oh! Interesting. I didn't catch that bit on the next line.

`Regexp.last_match` is not an approach I'd seen before! Super interesting.